@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -278,7 +278,9 @@ export class TileGeometryLoader {
             execute: this.prepare.bind(this, enabledKinds, disabledKinds),
             group: TileTaskGroups.CREATE,
             getPriority: this.getPriority.bind(this),
-            isExpired: this.discardNeedlessTile.bind(this, this.tile),
+            isExpired: () => {
+                return !this.tile.isVisible || this.tile.dataSource.isDetached();
```

Comment:
Have you checked if this maybe throws an error, because if I remember correctly a while ago there was an issue with Tile.isVisible accessing tile.datasource.mapView, so I think you might have to check for this.tile.dataSource.isDetached(); before this.tile.isVisible()

Reply:
Good spotting! I will fix.

I'm surprised my test where I detach the mapview doesn't pick this up however, because the `update` is called afterwards, and I would expect that is when the `isExpiring` method is called... will invistigate.